packages/react-native-sdk/docusaurus/docs/reactnative/03-core/08-native-permissions.mdx

```diff
@@ -3,83 +3,86 @@ id: native-permissions
+
+Once the component using this hook is mounted, we should see permissions being requested like below:
+
+![Preview of the final result](../assets/03-core/08-native-permissions/permissions.png)
+
+## Setup
+
+Ensure that relevant permissions are declared in your `AndroidManifest.xml` and `Info.plist` as mentioned in the [installation](../setup/installation/) guide. 
+
+Additionally, to request permissions in both platforms easily we will use the [`react-native-permissions`](https://github.com/zoontek/react-native-permissions) library. You can run the following command to install it:
+
+```bash title=Terminal
+yarn add react-native-permissions
+```
 
+#### iOS
+
+Additionally, for iOS, you need update your `package.json` by adding the permissions used in your app.
+
+```js title=package.json
+{
+  "reactNativePermissionsIOS": [
+    "Camera",
+    "Microphone",
+  ],
+}
+```
+
+After that, we need to run the following commands:
+
+```bash title=Terminal
+npx react-native setup-ios-permissions
+npx pod-install
+```
+
+## Step 1 - Add functions to update permissions to the SDK
+
+In this step, we create two functions called `androidProcessPermissions` and `iOSProcessPermissions`. These functions are responsible for informing the status of the permission grants to the SDK. 
+
+```ts title=src/utils/updatePermissions.ts
+import { PERMISSIONS, RESULTS, PermissionStatus } from 'react-native-permissions';
 import { StreamVideoRN } from '@stream-io/video-react-native-sdk';
 
-export const useSyncPermissions = () => {
-  useEffect(() => {
-    // Check, update and sync permissions after parent component mounts
-    requestAndUpdatePermissions();
-  }, []);
-  // useAppStateListener(handleOnForeground, () => {});
-};
 
-const requestAndUpdatePermissions = async () => {
-  if (Platform.OS === 'ios') {
-    // Request camera and mic permissions on iOS
-    const results = await requestMultiple([
-      PERMISSIONS.IOS.CAMERA,
-      PERMISSIONS.IOS.MICROPHONE,
-    ]);
-    // Sync the permissions with the Stream Video SDK
-    // This ensures that the SDK will subscribe to audio/video devices as preparation for a call.
-    iOSProcessResultsAndSetToConfig(results);
-    return;
-  } else if (Platform.OS === 'android') {
-    // Request camera, mic, bluetooth and notification permissions on Android
-    const results = await requestMultiple([
-      PERMISSIONS.ANDROID.CAMERA,
-      PERMISSIONS.ANDROID.RECORD_AUDIO,
-      PERMISSIONS.ANDROID.BLUETOOTH_CONNECT,
-      PERMISSIONS.ANDROID.POST_NOTIFICATIONS,
-    ]);
-    // Sync the camera and permissions with the Stream Video SDK
-    // This ensures that the SDK will subscribe to audio/video devices as preparation for a call.
-    androidProcessResultsAndSetToConfig(results);
-  }
-};
+// The extracted type of the return value of `requestMultiple`
+type Results = Awaited<ReturnType<typeof requestMultiple>>;
```

Comment:
Was trying the tutorial locally, but this seems to be unused in the updatePermissions.ts.

Can you check?

Reply:
yes, forgot to remove this line